### PR TITLE
Feature/xz 36/cleanup

### PR DESCRIPTION
**Pull request by bbrands02**

### CODE REVIEW

File: Service/XxllncToZGWZaakService.php [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+    } // end _construct
+
+    /**
+     * Set symfony style in order to output to the console.
+     *
+     * @param SymfonyStyle $io
+     *
+     * @return self
+     */
+    public function setStyle(SymfonyStyle $io): self
```
Comment (WilcoLouwerse) [inline Service/XxllncToZGWZaakService.php:77]: We should move to monolog later (at least at a todo for this)

File: Service/XxllncToZGWZaakService.php [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+    }// end setDefaultValues
+
+    /**
+     * Creates or updates a case to zaak.
+     *
+     * @param array $case Case from the Xxllnc API
+     *
+     * @return void|null
+     */
+    public function caseToZaak(array $case)
```
Comment (WilcoLouwerse) [inline Service/XxllncToZGWZaakService.php:272]: This is a big function, could be smaller

File: Service/XxllncToZGWZaakTypeService.php [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+        CallService $callService
+    ) {
+        $this->entityManager = $entityManager;
+        $this->synchronizationService = $synchronizationService;
+        $this->callService = $callService;
+
+        $this->objectRepo = $this->entityManager->getRepository(ObjectEntity::class);
+        $this->schemaRepo = $this->entityManager->getRepository(Schema::class);
+        $this->sourceRepo = $this->entityManager->getRepository(Source::class);
+        $this->mappingRepo = $this->entityManager->getRepository(Mapping::class);
```
Comment (WilcoLouwerse) [inline Service/XxllncToZGWZaakTypeService.php:54]: I don't like doing this^ (just my opinion), but can't come up with a reason why we shouldn't do this right now.

File: Service/XxllncToZGWZaakTypeService.php [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+
+        $this->skeletonIn = [
+            'handelingInitiator'   => 'indienen',
+            'beginGeldigheid'      => '1970-01-01',
+            'versieDatum'          => '1970-01-01',
+            'doel'                 => 'Overzicht hebben van de bezoekers die aanwezig zijn',
+            'versiedatum'          => '1970-01-01',
+            'handelingBehandelaar' => 'Hoofd beveiliging',
+            'aanleiding'           => 'Er is een afspraak gemaakt met een (niet) natuurlijk persoon',
+        ];
```
Comment (WilcoLouwerse) [inline Service/XxllncToZGWZaakTypeService.php:64]: skeletonIn feels like something we should configure in some .json file and not in a service variable or constructor

File: Service/XxllncToZGWZaakTypeService.php [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+     * @param SymfonyStyle $io
+     *
+     * @return self
+     */
+    public function setStyle(SymfonyStyle $io): self
+    {
+        $this->io = $io;
+
+        return $this;
+    }// end setStyle
```
Comment (WilcoLouwerse) [inline Service/XxllncToZGWZaakTypeService.php:79]: We should move to monolog later (at least at a todo for this)

File: Service/XxllncToZGWZaakTypeService.php [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+
+    /**
+     * Maps the statusTypen and rolTypen from xxllnc to zgw.
+     *
+     * @param array $caseType      This is the xxllcn casetype array.
+     * @param array $zaakTypeArray This is the ZGW ZaakType array.
+     *
+     * @return array $zaakTypeArray This is the ZGW ZaakType array with the added statustypen.
+     */
+    private function mapStatusAndRolTypen(array $caseType, array $zaakTypeArray): array
```
Comment (WilcoLouwerse) [inline Service/XxllncToZGWZaakTypeService.php:113]: Big function, could be smaller or more readable

File: Service/XxllncToZGWZaakTypeService.php [hunk 7]
```diff
@@ -0,0 +1,376 @@
...
+     * Creates or updates a casetype to zaaktype.
+     *
+     * @param array $caseType CaseType from the Xxllnc API
+     * @param bool  $flush    Do we need to flush here
+     *
+     * @var Synchronization
+     *
+     * @return void|null
+     */
+    public function caseTypeToZaakType(array $caseType, bool $flush = true)
```
Comment (WilcoLouwerse) [inline Service/XxllncToZGWZaakTypeService.php:281]: big function, could be smaller

File: Service/ZGWToXxllncZaakService.php [hunk 8]
```diff
@@ -0,0 +1,499 @@
...
+     */
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        CallService $callService
+    ) {
+        $this->entityManager = $entityManager;
+        $this->callService = $callService;
+
+        $this->schemaRepo = $this->entityManager->getRepository(Schema::class);
+        $this->sourceRepo = $this->entityManager->getRepository(Source::class);
```
Comment (WilcoLouwerse) [inline Service/ZGWToXxllncZaakService.php:43]: I don't like doing this^ (just my opinion), but can't come up with a reason why we shouldn't do this right now.

File: Service/ZGWToXxllncZaakService.php [hunk 9]
```diff
@@ -0,0 +1,499 @@
...
+     * @param SymfonyStyle $io
+     *
+     * @return self
+     */
+    public function setStyle(SymfonyStyle $io): self
+    {
+        $this->io = $io;
+
+        return $this;
+    }// end setStyle
```
Comment (WilcoLouwerse) [inline Service/ZGWToXxllncZaakService.php:58]: We should move to monolog later (at least at a todo for this)

File: Service/ZGWToXxllncZaakService.php [hunk 10]
```diff
@@ -0,0 +1,499 @@
...
+
+    public function updateZaakWithEigenschapHandler(?array $data = [], ?array $configuration = [])
+    {
+        isset($this->io) && $this->io->success('updateZaakWithEigenschapHandler triggered');
+        var_dump('updateZaakWithEigenschapHandler triggered');
```
Comment (WilcoLouwerse) [inline Service/ZGWToXxllncZaakService.php:63]: var_dump

File: Service/ZGWToXxllncZaakService.php [hunk 11]
```diff
@@ -0,0 +1,499 @@
...
+        $caseId = end($caseId);
+        $zaakObject = $this->entityManager->find(ObjectEntity::class, $caseId);
+        $zaakObjectArray = $zaakObject->toArray();
+
+        $this->getRequiredGatewayObjects();
+
+
+        if (!isset($zaakObjectArray['zaaktype'])) {
+            // throw new \Exception('No zaaktype set on zaak');
+        var_dump('test1 ');
```
Comment (WilcoLouwerse) [inline Service/ZGWToXxllncZaakService.php:76]: var_dump

File: Service/ZGWToXxllncZaakService.php [hunk 12]
```diff
@@ -0,0 +1,499 @@
...
+
+            return ['response' => $data];
+        }
+        
+
+        if (!isset($zaakObjectArray['zaaktype']['_self']['id'])) {
+            // throw new Exception('ZaakType id not found on Zaak object');
+        var_dump('test2 ');
```
Comment (WilcoLouwerse) [inline Service/ZGWToXxllncZaakService.php:84]: var_dump

File: Service/ZGWToXxllncZaakService.php [hunk 13]
```diff
@@ -0,0 +1,499 @@
...
+
+            return ['response' => $data];
+        }
+
+        $zaakTypeId = $zaakObjectArray['zaaktype']['_self']['id'];
+        $zaakTypeObject = $this->entityManager->find(ObjectEntity::class, $zaakTypeId);
+        $casetypeId = $zaakTypeObject->getSynchronizations()[0]->getSourceId() ?? null;
+        // Return here cause if the zaaktype is created through this gateway, we cant sync it to xxllnc because it doesn't exist there
+        if (!isset($casetypeId)) {
+            var_dump('test3 ');
```
Comment (WilcoLouwerse) [inline Service/ZGWToXxllncZaakService.php:94]: var_dump

File: Service/ZGWToXxllncZaakService.php [hunk 14]
```diff
@@ -0,0 +1,499 @@
...
+    
+            return ['response' => $data];
+        }
+
+        if (!isset($zaakObjectArray['_self']['id'])) {
+            // throw new \Exception('No id on zaak'); // meaning it didnt properly save in the gateway
+            var_dump('test4 ');
```
Comment (WilcoLouwerse) [inline Service/ZGWToXxllncZaakService.php:101]: var_dump

File: Service/ZGWToXxllncZaakService.php [hunk 15]
```diff
@@ -0,0 +1,499 @@
...
+
+            return ['response' => $data];
+        }
+
+        $zaakArrayObject = $this->entityManager->find('App:ObjectEntity', $zaakObjectArray['_self']['id']);
+
+        if (!isset($zaakArrayObject)) {
+            // throw new \Exception('ZGW Zaak not found with id: ' . $this->data['_self']['id']);
+            var_dump('test5 ');
```
Comment (WilcoLouwerse) [inline Service/ZGWToXxllncZaakService.php:110]: var_dump

File: Service/ZGWToXxllncZaakService.php [hunk 16]
```diff
@@ -0,0 +1,499 @@
...
+     *
+     * @return array $xxllncZaakArray This is the Xxllnc Zaak array with the added eigenschappen.
+     */
+    private function mapPostEigenschappen(array $xxllncZaakArray, array $zaakArrayObject, array $zaakTypeEigenschappen): array
+    {
+        $eigenschapIds = [];
+        foreach ($zaakTypeEigenschappen as $eigenschap) {
+            $eigenschapIds[] = $eigenschap->getId()->toString();
+        }
+        dump($eigenschapIds);
```
Comment (WilcoLouwerse) [inline Service/ZGWToXxllncZaakService.php:136]: dump

File: Service/ZGWToXxllncZaakService.php [hunk 17]
```diff
@@ -0,0 +1,499 @@
...
+    //     if (!$zaakObject instanceof ObjectEntity) {
+    //         throw new \Exception('Zaak object not found with id:' . $this->data['zaak']['_self']['id']);
+    //     }
+
+    //     $zaakArray = $zaakObject->toArray();
+
+    //     $xxllncZaakArray = $this->mapZGWToXxllncZaak($casetypeId, $zaakTypeObject, $zaakArray, $xxllncZaakPostEntity, false);
+
+    //     return ['response' => $xxllncZaakArray, 'entity' => $xxllncZaakPostEntity->getId()->toString()];
+    // }
```
Comment (WilcoLouwerse) [inline Service/ZGWToXxllncZaakService.php:296]: Lot of commented out code here, why is this still here? remove or add todo

File: Service/ZGWToXxllncZaakService.php [hunk 18]
```diff
@@ -0,0 +1,499 @@
...
+
+    /**
+     * Maps zgw zaak to xxllnc case.
+     *
+     * @param string       $caseTypeId     CaseTypeID as in xxllnc.
+     * @param ObjectEntity $zaakTypeObject ZGW ZaakType object
+     *
+     * @return array $this->data Data which we entered the function with
+     */
+    public function mapZGWToXxllncZaak(string $casetypeId, ObjectEntity $zaakTypeObject, array $zaakArrayObject)
```
Comment (WilcoLouwerse) [inline Service/ZGWToXxllncZaakService.php:353]: big function, could be smaller

File: Service/ZGWToXxllncZaakService.php [hunk 19]
```diff
@@ -0,0 +1,499 @@
...
+
+    /**
+     * Creates or updates a ZGW Zaak from a xxllnc casetype with the use of mapping.
+     *
+     * @param ?array $data          Data from the handler where the xxllnc casetype is in.
+     * @param ?array $configuration Configuration from the Action where the Zaak entity id is stored in.
+     *
+     * @return array $this->data Data which we entered the function with
+     */
+    public function zgwToXxllncZaakHandler(?array $data = [], ?array $configuration = []): array
```
Comment (WilcoLouwerse) [inline Service/ZGWToXxllncZaakService.php:450]: big function, could be smaller

File: Schema/xxllnc-zaak-post.json [hunk 20]
```diff
@@ -1,6 +1,6 @@
 {
     "$id": "https://development.zaaksysteem.nl/schema/xxllnc.zaakPost.schema.json",
-    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$schema": "https://docs.commongateway.nl/schemas/Schema.schema.json",
```
Comment (WilcoLouwerse) [inline Schema/xxllnc-zaak-post.json:3]: https://docs.commongateway.nl/schemas/Entity.schema.json

File: Schema/xxllnc-zaak-requestor.json [hunk 21]
```diff
@@ -1,6 +1,6 @@
 {
     "$id": "https://development.zaaksysteem.nl/schema/xxllnc.zaakRequestor.schema.json",
-    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$schema": "https://docs.commongateway.nl/schemas/Schema.schema.json",
```
Comment (WilcoLouwerse) [inline Schema/xxllnc-zaak-requestor.json:3]: https://docs.commongateway.nl/schemas/Entity.schema.json

**Review by WilcoLouwerse — CHANGES_REQUESTED** [review]

File: Service/XxllncToZGWZaakService.php [hunk 22]
```diff
@@ -0,0 +1,378 @@
...
+        $this->entityManager = $entityManager;
+        $this->synchronizationService = $synchronizationService;
+        $this->callService = $callService;
+        $this->xxllncToZGWZaakTypeService = $xxllncToZGWZaakTypeService;
+
+        $this->objectRepo = $this->entityManager->getRepository(ObjectEntity::class);
+        $this->schemaRepo = $this->entityManager->getRepository(Schema::class);
+        $this->sourceRepo = $this->entityManager->getRepository(Source::class);
+        $this->synchronizationRepo = $this->entityManager->getRepository(Synchronization::class);
+        $this->mappingRepo = $this->entityManager->getRepository(Mapping::class);
```
[the remark this answers is not in the document]
Comment (rjzondervan) [reply Service/XxllncToZGWZaakService.php:58]: Consistency, fine for now, but please use 'App:{{class}}'.

File: Service/XxllncToZGWZaakService.php [hunk 23]
```diff
@@ -0,0 +1,378 @@
...
+
+        $this->skeletonIn = [
+            'verantwoordelijkeOrganisatie' => '070124036',
+            'betalingsindicatie'           => 'geheel',
+            'betalingsindicatieWeergave'   => 'Bedrag is volledig betaald',
+            'laatsteBetaalDatum'           => '15-07-2022',
+            'archiefnominatie'             => 'blijvend_bewaren',
+            'archiefstatus'                => 'nog_te_archiveren',
+        ];
```
[the remark this answers is not in the document]
Comment (rjzondervan) [reply Service/XxllncToZGWZaakService.php:68]: Should be in the mapping

File: Service/XxllncToZGWZaakService.php (hunk 22 above)
[the remark this answers is not in the document]
Comment (WilcoLouwerse) [reply Service/XxllncToZGWZaakService.php:58]: ->getRepository('App:Entity');
@bbrands02

File: Service/XxllncToZGWZaakService.php (hunk 23 above)
[the remark this answers is not in the document]
Comment (WilcoLouwerse) [reply Service/XxllncToZGWZaakService.php:68]: Add a todo for this

File: Service/XxllncToZGWZaakTypeService.php (hunk 3 above)
Comment (WilcoLouwerse) [reply Service/XxllncToZGWZaakTypeService.php:54]: ->getRepository('App:Entity');
@bbrands02

File: Service/ZGWToXxllncZaakService.php (hunk 8 above)
Comment (WilcoLouwerse) [reply Service/ZGWToXxllncZaakService.php:43]: ->getRepository('App:Entity');
@bbrands02

File: Service/XxllncToZGWZaakTypeService.php (hunk 4 above)
Comment (WilcoLouwerse) [reply Service/XxllncToZGWZaakTypeService.php:65]: Should be in the mapping
Add a todo for this

File: Service/XxllncToZGWZaakService.php (hunk 2 above)
Comment (WilcoLouwerse) [reply Service/XxllncToZGWZaakService.php]: Add a todo for this

File: Service/XxllncToZGWZaakTypeService.php (hunk 4 above)
Comment (bbrands02) [reply Service/XxllncToZGWZaakTypeService.php:65]: image
Git loopt achter

File: Service/ZGWToXxllncZaakService.php (hunk 17 above)
Comment (bbrands02) [reply Service/ZGWToXxllncZaakService.php:285]: image
git loopt achter ?

File: Service/XxllncToZGWZaakService.php (hunk 23 above)
[the remark this answers is not in the document]
Comment (bbrands02) [reply Service/XxllncToZGWZaakService.php:68]: image
git loopt achter

File: Service/XxllncToZGWZaakService.php (hunk 2 above)
Comment (bbrands02) [reply Service/XxllncToZGWZaakService.php]: image
opgeschoond

**Review by WilcoLouwerse — APPROVED** [review]

**Review by rjzondervan — APPROVED** [review]

**Pull request merged**